### ISSUE TITLE
[BottomSheetModal], a prop should allow elements below the sheet to not be accessible and the sheet should receive focus when presented

### ISSUE THREAD

**Issue by jsamr** [gorhom/react-native-scroll-bottom-sheet#687]
# Feature Request

I am implementing a feature for an app provided by the French state and accessibility is a top priority concern. The idea would be to reproduce the behavior of React Native `Modal` component whereby elements below the modal are not accessible.

The feature would be opt-in with a prop, `accessibiltiyHideSiblings` or something along those lines.

## Why it is needed

Screen reader users are confused when elements below the sheet are accessible. They are also confused when the focus is not passed to the modal, quoting Shopify accessibility team which shares conclusions of a study they made:

> When opening a modal window, focus must be placed on or inside of the modal window in order to bring context awareness to the user. Otherwise, focus remains on the activator control and the user may not be aware of or be able to easily reach the modal content.

Source: [Creating Accessible React Native Apps](https://www.shopify.com/partners/blog/react-native-accessibility)

## Possible implementation

The `<BottomSheetModalProvider>` which AFAIK acts like a portal could do the following:

- Hide siblings when modal is presented;
- Provide focus to the sheet when modal becomes presented.

### Code sample

``` jsx
function BottomSheetModalProvider() {
  // ...
  const isModalVisible = true;
  return (
    <>
      {/* Descendants */}
      <View
        importantForAccessibility={
          isModalVisible ? "no-hide-descendents" : "yes"
        }
      >
        {children}
      </View>

      <View
        importantForAccessibility={
          isModalVisible ? "yes" : "no-hide-descendents"
        }
        accessibilityViewIsModal={isModalVisible}
      >
        {/* Bottom sheet portal */}
      </View>
    </>
  );
}
```

This snippet does not include the second point (provide focus). This point can be handled by providing to the `View` which should receive focus the below function:

```jsx
/**
 * A function to pass as `onLayout` prop to focus on mount for screen readers.
 */
export function focusOnLayoutForScreenReaders(e: LayoutChangeEvent) {
  //@ts-expect-error - e.target is mistyped (it's actually a react element).
  const node = findNodeHandle(e.target);
  if (node) {
    AccessibilityInfo.setAccessibilityFocus(node);
    // Fix a bug on Android, see https://github.com/facebook/react-native/issues/30097
    AccessibilityInfo.setAccessibilityFocus(node);
  }
}
```

##### `accessibilityViewIsModal` (iOS)

> A Boolean value that indicates whether VoiceOver **ignores the accessibility elements within views that are siblings of the element**.

Quite self-explicit!

[Apple documentation](https://developer.apple.com/documentation/objectivec/nsobject/1615089-accessibilityviewismodal)

##### `importantForAccessibility` (Android)

> Describes whether or not this view is important for accessibility. If it is important, the view fires accessibility events and is reported to accessibility services that query the screen. Note: While not recommended, an accessibility service may decide to ignore this attribute and operate on all views in the view tree. 
> **yes**: The view is important for accessibility.
> **no-hide-descendants**: The view is not important for accessibility, nor are any of its descendant views.

By using "no-hide-descendants" on the only sibling of the portal, we are effectively reproducing the behavior of `accessibilityViewIsModal` on iOS.

[Google documentation](https://developer.android.com/reference/android/view/View#attr_android:importantForAccessibility)

**Comment by filipef101** [comment]
@jsamr Did you happen to make any attempt at this?

**Comment by jsamr** [comment]
@filipef101 Nope, I just mentioned this ticket in our issue tracker.

**Comment by antonandreyev** [comment]
@gorhom is it possible to reopen this? This part 
>Screen reader users are confused when elements below the sheet are accessible.

Is still extremely valid. 


This part 
> Provide focus to the sheet when modal becomes presented.

Is easy to fix on user side like this:
```tsx
import { AccessibilityInfo, findNodeHandle } from "react-native";

export const setInitA11yFocus = (element: React.Component | null) => {
  if (element == null) return;
  const elementId = findNodeHandle(element);
  if (elementId) {
    // have to do 2 times because of RN bug
    AccessibilityInfo.setAccessibilityFocus(elementId);
    AccessibilityInfo.setAccessibilityFocus(elementId);
  }
};

// ...
return (
  <BottomSheetModal
   // ...
  >
    <View accessible accessibilityRole="header" ref={setInitA11yFocus}>
        {/* first item, e.g. heading, in bottom sheet content */}
      </View>
    {/* ... */}
  </BottomSheetModal>);
```

And since it somewhat depends on the content, I don't think it makes sense to implement this on the library level. Maybe instead provide some instructions or even API

**Comment by scottgrunerud** [comment]
Can we re-open this issue as it still exists?
